packages/ckeditor5-coremedia-example-data/src/InitExamples.ts

```diff
@@ -0,0 +1,201 @@
+}
+
+const initExamplesUi = (parent: ParentNode): ExamplesUiElements => {
+  const dataList = createDataListElement();
+  const input = createInputWithDataFrom(dataList);
+  const label = createLabelFor(input);
+  const reload = createReloadButton();
+  const clear = createClearButton();
+
+  parent.append(label, input, dataList, reload, clear);
+
+  return { input, dataList, reload, clear };
+};
+
+const addExampleOptions = (
+  dataList: HTMLDataListElement,
+  defaultKey: string | undefined,
+  exampleKeys: string[],
+): void => {
+  // Now add all examples
+  for (const exampleKey of exampleKeys.sort()) {
+    const option = document.createElement("option");
+    option.textContent = exampleKey;
+    option.value = exampleKey;
+    option.defaultSelected = exampleKey === defaultKey;
+    dataList.appendChild(option);
+  }
+};
+
+/**
+ * Initializes the examples.
+ *
+ * @param config - configuration for examples
+ */
+export const initExamples = (config: ExamplesConfig): void => {
+  const { id = "examples", default: defaultExampleKey, examples, onChange } = config;
+  const examplesContainer = document.getElementById(id);
+
+  if (!examplesContainer) {
+    throw new Error(`Cannot locate examples container with ID "${id}".`);
+  }
+
+  const { input, dataList, reload, clear } = initExamplesUi(examplesContainer);
+
+  // On change, set the data – or show an error if data are unknown.
+  input.addEventListener("change", () => {
+    const newValue = input.value;
+    if (examples.hasOwnProperty(newValue)) {
+      input.classList.remove("error");
+      const data = examples[newValue];
+      console.log("Setting example data.", { [newValue]: data });
+      onChange(data);
+      input.blur();
+    } else {
+      input.classList.add("error");
+      input.select();
+    }
+  });
+
+  // Init the reload-button, to also listen to the value of example input field.
+  reload.addEventListener("click", () => {
+    const newValue = input.value;
+    if (examples.hasOwnProperty(newValue)) {
+      input.classList.remove("error");
+      const data = examples[newValue];
+      console.log("Resetting example data.", { [newValue]: data });
+      onChange(data);
+      input.blur();
+    }
+  });
+
+  clear.addEventListener("click", () => {
+    input.blur();
+    console.log("Clearing data.");
+    onChange("");
+  });
+
+  addExampleOptions(dataList, defaultExampleKey, Object.keys(examples));
+
+  if (defaultExampleKey) {
+    input.value = defaultExampleKey;
+    if (examples.hasOwnProperty(defaultExampleKey)) {
+      input.classList.remove("error");
+      const data = examples[defaultExampleKey];
+      console.log("Setting default example data.", { [defaultExampleKey]: data });
+      onChange(data);
+    } else {
+      input.classList.add("error");
+      console.error(`Invalid default example key given: "${defaultExampleKey}".`);
+    }
+  }
+};
+
+/**
+ * Examples configuration.
+ */
+export interface ExamplesConfig {
```

Comment:
Short: Won't Do.

I doubt, that it will reduce complexity. The complexity will raise just somewhere else then: Extend TSDoc to list all config options, and a longer, wrapped type on function declaration.

Still, we may decide differently with ease at any time, as the package is private anyway.